src/game_objects/construct.cpp

```diff
@@ -12,80 +12,128 @@ Construct::Construct(int x, int z, int id) : GameObject(id) {
+// Activated when a player presses A on it, graphical
+void Chest::c_interact_trigger() {
     anim_player.set_anim("open");
     anim_player.play();
 }
 
 void Chest::setup_model() {
     attach_model(AssetLoader::get_model("chest_good"));
-    transform.set_scale({ 0.005f, 0.005f, 0.005f });
+    transform.set_scale({ 0.006f, 0.006f, 0.006f });
+}
+
+/************SPIKES***************/
+Spikes::Spikes(int x, int z) : Construct(x, z) {
+    tag = "SPIKES";
+
+    //Creates Rigid Body
+    events::RigidBodyData rigid;
+    rigid.object = this;
+    rigid.shape = hit_box;
+    rigid.is_ghost = true;
+    events::add_rigidbody_event(rigid);
+
+    // Sets Initial Position
+    set_position({ x, 0.0f, z });
+}
+
+Spikes::Spikes(int x, int z, int id) : Construct(x, z, id) {
+    tag = "SPIKES";
+    // Sets Initial Position (Transform and Rigidbody at same time)
+    set_position({ x, 0.0f, z });
+}
+
+void Spikes::s_on_collision(GameObject *other) {
+    Player *player = dynamic_cast<Player*>(other);
+    if (player) {
+        events::dungeon::network_collision_event(id);
+    }
+}
+
+void Spikes::c_on_collision() {
+    std::cerr << "TEST" << std::endl;
+    anim_player.set_anim("trigger");
+    anim_player.play();
+}
+
+void Spikes::setup_model() {
+    attach_model(AssetLoader::get_model("spikes"));
+    transform.set_scale({ 0.4f, 0.4f, 0.4f });
 }
 
+/************GOAL***************/
 Goal::Goal(int x, int z) : Construct(x, z) {
-    transform.set_scale(0.006f, 0.006f, 0.006f);
     tag = "GOAL";
 
-    events::RigidBodyData rigid = {
-        glm::vec3(x,0.5f,z), //position
-        0, //mass
-        goal_hit_box, //btshape
-        glm::vec3(0,0,0), //inertia
-        this, //the gameobject
-        true, // is a ghost object
-    };
+    events::RigidBodyData rigid;
+    rigid.object = this;
+    rigid.shape = goal_hit_box;
+    rigid.is_ghost = true;
     events::add_rigidbody_event(rigid);
 
     notify_on_collision = true;
+
+    set_position({ x, 0.5f, z });
 }
 
 Goal::Goal(int x, int z, int id) : Construct(x, z, id) {
-    transform.set_scale(0.006f, 0.006f, 0.006f);
     tag = "GOAL";
+    set_position({ x, 0.5f, z });
 }
 
 void Goal::setup_model() {
     attach_model(AssetLoader::get_model("warp"));
+    transform.set_scale(0.006f, 0.006f, 0.006f);
     model->set_shader(&ShaderManager::anim_unlit);
     anim_player.set_speed(1.0f);
     anim_player.set_anim("spin");
     anim_player.set_loop(true);
     anim_player.play();
 }
 
-void Goal::on_collision(GameObject *other) {
+void Goal::s_on_collision(GameObject *other) {
     Player *player = dynamic_cast<Player*>(other);
     if (player) {
         events::dungeon::network_collision_event(id);
         events::dungeon::player_finished_event(player->get_id());
     }
 }
 
-void Goal::on_collision_graphical() {
-}
-
-void Goal::update_this() {
-    anim_player.update();
+void Goal::c_on_collision() {
+    if (anim_player.check_paused()) {
+        anim_player.play();
+    }
```

Comment:
Oh, that was just a mistake while I was rebasing. Didn't realize that, will fix.